swfo/h5utils.py

```diff
@@ -0,0 +1,93 @@
+from typing import Optional, List, Dict, Union
+from functools import partial
+import yaml
+
+import h5py
+
+VLEN_STRING = h5py.special_dtype(vlen=str)
+
+
+yaml_dumper = partial(yaml.dump, indent=4, default_flow_style=False)
+
+
+def _get_next_md_id(h5_container: h5py.Group, group_prefix: str) -> int:
+    """
+    Returns the next incremental ID for metadata versioning
+    """
+    ids = [0]
+    def _append_id(h5_key):
+        ids.append(int(h5_key[1:]))
+
+    group = h5_container.get(group_prefix)
+    if group:
+        group.visit(_append_id)
+
+    return max(ids) + 1
+
+
+def write_h5_md(
+        h5_container: h5py.Group,
```

Comment:
A minor concern here is using a different label for an H5 Group.  Can we use h5_group as the variable name?